### PR TITLE
fix: Apply SP error fix to scale updates

### LINKED ISSUE #518: SP: Should we apply the underflow fix to the case where we update the scale?
Address the TODO from #438

### PR DESCRIPTION
**Pull request by bingen**
Fixes #518

### CODE REVIEW

**Review by danielattilasimon — APPROVED** [review]
Looks good to me, and it has been running on the fuzzer for a week without issues, but I would wait for @RickGriff's approval, too.

**Comment by bingen** [comment]
Replaced by #552

**Pull request closed without merging**
